### PR TITLE
Refactor product status command

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | The active field (managing online/offline status) should not be handled via options tab anymore nor the options command This PR includes following modifications:<br>- clean active from options command and query<br>- handle active via its dedicated command<br>- improve management of indexation based on active status and visibility configuration<br>- update command builders<br>- update behat tests
| Type?             | refacto
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | CI tests green
| Possible impacts? | ~

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/26449)

### CODE REVIEW

File: src/Adapter/Product/CommandHandler/UpdateProductStatusHandler.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        $product = $this->productRepository->get($command->getProductId());
+        $initialState = (bool) $product->active;
+        $product->active = $command->getEnable();
+        $this->productRepository->partialUpdate(
+            $product,
+            ['active'],
+            CannotUpdateProductException::FAILED_UPDATE_STATUS
+        );
+
+        // If status changed we need to update its indexes (we check if it is necessary because index build can be
```
Comment (matks) [inline src/Adapter/Product/CommandHandler/UpdateProductStatusHandler.php:78]: Even if operation was not expensive, clearing a cache or an index should only be done when necessary 👍

File: src/Adapter/Product/Update/ProductIndexationUpdater.php [hunk 2]
```diff
@@ -28,28 +28,99 @@
...
+    {
+        $this->isSearchIndexationOn = $isSearchIndexationOn;
+    }
+
+    /**
+     * @param Product $product
+     *
+     * @return bool
+     */
+    public function isVisibleOnSearch(product $product): bool
```
Comment (matks) [inline src/Adapter/Product/Update/ProductIndexationUpdater.php]:
```suggestion
    public function isVisibleOnSearch(Product $product): bool
```

Comment (jolelievre) [reply src/Adapter/Product/Update/ProductIndexationUpdater.php]: Done

**Review by matthieu-rolland — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by jolelievre** [comment]
Thank me

**Pull request merged**